src/tools/pub/dependencies.md

```diff
@@ -237,93 +235,101 @@ If it's an unknown identifier, the dependency is always considered unsatisfied.
+
 dependencies:
-  path: ^1.3.0
-  collection: ^1.1.0
-  string_scanner: ^0.1.2
+  transmogrify:
+    hosted:
+      name: transmogrify
+      url: https://some-package-server.com
+    # This package must use a 1.x version of transmogrify starting with 1.4.
+    version: ^1.4.0
 ```
 
+To learn more about pub's version system, see the [package versioning page][].
+
+[package versioning page]: {{site.url}}/tools/pub/versioning#semantic-versions
+
 ### Traditional syntax
 
-A version constraint that uses _traditional syntax_
-is a series of the following:
+A version constraint that uses the traditional syntax can use any
+of the following values:
 
-`any`
-: The string `any` allows any version. This is equivalent to an empty
-  version constraint, but is more explicit. **Although `any` is allowed,
-  we don't recommend it.**
+| **Value** |                **Allows**               | **Use?** |                                                                        **Notes**                                                                        |
+|:---------:|:----------------------------------------|:--------:|:--------------------------------------------------------------------------------------------------------------------------------------------------------|
+|   `any`   | All versions                            |    No    | Serves as a explicit declaration of empty version constraint.                                                                                           |
+|  `1.2.3`  | Only the given version                  |    No    | Limits adoption of your package due the additional limits it places on apps that use your package.                                                      |
+| `>=1.2.3` | Given version or later                  |    Yes   |                                                                                                                                                         |
+|  `>1.2.3` | Versions later than the given version   |    No    |                                                                                                                                                         |
+| `<=1.2.3` | Given version or earlier                |    No    |                                                                                                                                                         |
+|  `<1.2.3` | Versions earlier than the given version |    No    | Use this when you know an upper bound version that _doesn't_ work with your package. This version might be the first to introduce some breaking change. |
+{:.table}
 
-`1.2.3`
-: A concrete version number pins the dependency to only allow that
-    <em>exact</em> version. Avoid using this when you can because it can cause
-    version lock for your users and make it hard for them to use your package
-    along with other packages that also depend on it.
+You can specify any combination of version values as their ranges intersect.
```

Comment:
What it is exactly we mean here?

Reply:
I have a better explanation in the updated copy.